packages/swapper/src/swappers/cow/CowSwapper.ts

```diff
@@ -37,9 +52,8 @@ export class CowSwapper implements Swapper {
     throw new Error('CowSwapper: getTradeQuote unimplemented')
   }
 
-  getUsdRate(input: Pick<Asset, 'symbol' | 'assetId'>): Promise<string> {
-    console.info(input)
-    throw new Error('CowSwapper: getUsdRate unimplemented')
+  async getUsdRate(input: Pick<Asset, 'symbol' | 'assetId'>): Promise<string> {
```

Comment:
looking at this further, you can simplify this more

https://github.com/shapeshift/lib/blob/main/packages/swapper/src/api.ts#L192

just pass the full asset in as the input, like the base class does, and then you can pull the precision from the asset itself, and you don't need to inject all the assets, just to take the asset id, to look up the full asset again...

Reply:
Oops yes ! done !